src/receiver-manager.js

```diff
@@ -202,8 +202,20 @@ class ReceiverManager {
   }
 
   _onPlayerLoadCompleteEvent(): void {
-    this._player.load();
-    this._player.ready().then(() => this._tracksManager.setInitialTracks());
+    const loadPlayerAndSetInitialTracks = () => {
+      this._logger.debug('Load player and set initial tracks');
+      this._player.load();
+      this._player.ready().then(() => this._tracksManager.setInitialTracks());
+    };
+    this._logger.debug('Player load complete');
+    if (this._adsManager.adBreak()) {
+      this._logger.debug('Ad break - load player and set initial tracks later');
```

Comment:
the debug messages seem a bit off, they are in context of the bug but no in context off the actual timeline of events.
You can print the fact that initial track setup was done in `setInitialTracks` and if you already do then this is really not necessary, in both debug logs you added.